tests/gen/README.md

```diff
@@ -0,0 +1,156 @@
+# Automatic Documents Generation 
+
+`AutoGenerate` and `AutoGenerateFromSDL` are a highly versatile function designed for dynamic document generation, perfect for testing and simulation purposes. 
+
+`AutoGenerateFromSDL` creates documents based on a specified schema, allowing for extensive customization of data generation. 
+
+The function generates documents adhering to a defined schema and it's configuration.
+It interprets the types and relationships within the schema to create realistic, interconnected data structures.
```

Comment:
todo: I think `defined schema and` and `relationships within the schema to` are incorrect, and should really read `defined collection and` and `relationships within the collection to`.

The distinction is still a bit blurry atm, but we are planning moving relation definitions off the schema and onto the collection (leaving only the primary side field on schema).

Reply:
> but we are planning moving relation definitions off the schema

Not 100% accurate, the decision was to not require *both* sides of the schema relation to be defined as I understood it. 
At the moment, we define the primary and secondary side of a realtionship, which causes some confusion and is overall a bad DX/UX. My understanding of the discussion on this was that we would only require adding a relationship to the primary side that actually depends/needs it, and then on the collection definition on the secondary side create a "fake" field so you can still query from both sides.

Reply:
> My understanding of the discussion on this was that we would only require adding a relationship to the primary side that actually depends/needs it, and then on the collection definition on the secondary side create a "fake" field so you can still query from both sides.

Yes, this is larger what I meant - I just don't see the primary field as being a relationship definition, just a type declaration that points 'somewhere else'.